optimum/habana/peft/layer.py

```diff
@@ -29,3 +31,41 @@ def GaudiLoraLayerLinearForward(self, x: torch.Tensor, *args: Any, **kwargs: Any
 
     result = result.to(previous_dtype)
     return result
+
+class LoRALinear:
+    def __init__(self, module):
+        has_bias = module.bias is not None
+        self.module = module
+        self.module.te_linear =  te.Linear(module.in_features, module.out_features, bias=has_bias, params_dtype=module.weight.dtype, skip_weight_param_allocation=True)
+
+    def _linear(self, input: torch.Tensor) -> torch.Tensor:
+        return self.module.te_linear(input, transpose(self.module.weight, self.module.fan_in_fan_out), bias=self.module.bias)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
```

Comment:
Better to align the function definition to forward(self, x: torch.Tensor, *args: Any, **kwargs: Any) -> torch.Tensor:

https://github.com/HabanaAI/optimum-habana-fork/pull/91/files#diff-3491044d2e0fd212ccdd50851ff639d6f0b4fba7d58f38bc92de53b5e1f56713R8

Reply:
Found it. GaudiLoraLayerLinearForward is based on old version.